force_wfmanager/utils/variable_names_registry.py

```diff
@@ -2,221 +2,326 @@
 
 from traits.api import (
     HasStrictTraits, List, Instance, on_trait_change, Property,
-    cached_property, Dict, Tuple)
+    cached_property, Dict, Tuple, HasTraits, Int, Unicode)
 
-from force_bdss.api import Identifier, Workflow
+from force_bdss.api import (
+    Identifier, Workflow, BaseDataSourceModel, OutputSlotInfo,
+    InputSlotInfo, VerifierError
+)
 from force_bdss.local_traits import CUBAType
 
 log = logging.getLogger(__name__)
 
 
-class VariableNamesRegistry(HasStrictTraits):
-    """ Class used for listening to the structure of the Workflow in order to
-    check the available variables that can be used as inputs for each layer
-    """
+class Variable(HasTraits):
+    """Class used to store UI input and output information from DataSources
+    as individual variables. These are then selected by the MCO optimiser as
+    possible parameters or KPIs"""
 
     # -------------------
     # Required Attributes
     # -------------------
 
-    #: Workflow model
-    workflow = Instance(Workflow, allow_none=False)
+    # CUBA type of Variable
+    type = CUBAType()
 
     # ------------------
     # Regular Attributes
     # ------------------
 
-    #: For each execution layer, there will be a list of (name, type) pairs
-    #: representing the input variables produced by that execution layer
-    data_source_input = List(Tuple(Identifier, CUBAType))
+    # Layer index at which variable is generated
+    layer = Int
 
-    #: For each execution layer, there will be a list of (name, type) pairs
-    #: representing the output variables produced by that execution layer
-    data_source_output = List(Tuple(Identifier, CUBAType))
+    # DataSource that contains Variable as an output
+    origin = Instance(BaseDataSourceModel)
```

Comment:
At the moment this trait is unused, but would be handy for creating a graphical representation of the execution layer process, since we have a reference of which `DataSources` are hooked up to each other